src/main/java/com/example/controllers/DashboardController.java

```diff
@@ -0,0 +1,206 @@
+import org.springframework.web.bind.annotation.ResponseBody;
+import org.springframework.web.multipart.MultipartFile;
+import org.springframework.web.servlet.mvc.support.RedirectAttributes;
+
+@Controller
+@RequestMapping(value = "/dashboard")
+public class DashboardController {
+
+	@Autowired
+	AccountDao accountDao;
+
+	@Autowired
+	CashAccountDao cashaccountDao;
+
+	@Autowired
+	CreditAccountDao creditaccountDao;
+
+	@Autowired
+	StorageFacade storageFacade;
+
+	@RequestMapping
+	public String activity(final Model model, final Principal principal) {
+
+		Account account = accountDao.findUsersByUsername(principal.getName()).get(0);
+
+		List<CashAccount> cashAccounts = cashaccountDao.findCashAccountsByUsername(principal.getName());
+		List<CreditAccount> creditAccounts = creditaccountDao.findCreditAccountsByUsername(principal.getName());
+
+		model.addAttribute("account", account);
+		model.addAttribute("cashAccounts", cashAccounts);
+		model.addAttribute("creditAccounts", creditAccounts);
+		return "dashboard";
+	}
+
+	@RequestMapping(value = "/userDetail", method = RequestMethod.GET)
+	public String userDetail(final Model model, final Principal principal, @RequestParam(value = "username") final String username) {
+
+		Account account = accountDao.findUsersByUsername(username).get(0);
+		List<CreditAccount> creditAccounts = creditaccountDao.findCreditAccountsByUsername(principal.getName());
+
+		model.addAttribute("creditAccounts", creditAccounts);
+		model.addAttribute("account", account);
+		model.addAttribute("accountMalicious", account);
+
+		return "userDetail";
+	}
+
+	@RequestMapping(value = "/userDetail/avatar", method = RequestMethod.GET)
+	public void getImage(final HttpServletResponse response, @RequestParam(value = "image") final String image) {
+
+		File file = storageFacade.exists(image) ? storageFacade.load(image) : storageFacade.load("avatar.png");
+		try {
+			writeResponse(new FileInputStream(file), response.getOutputStream());
+		}
+		catch (FileNotFoundException e) {
+			e.printStackTrace();
+		}
+		catch (IOException e) {
+			e.printStackTrace();
+		}
+	}
+
+	@RequestMapping(value = "/userDetail/avatar/update", method = RequestMethod.POST)
+	public String updateAvatar(@RequestParam("imageFile") final MultipartFile imageFile, final Principal principal,
+			final RedirectAttributes redirectAttributes) {
+		if (!imageFile.isEmpty()) {
+			try {
+				storageFacade.save(imageFile.getInputStream(), principal.getName() + ".png"); 
+			}
+			catch (IOException e) {
+				e.printStackTrace();
+			}
+		}
+
+		redirectAttributes.addAttribute("username", principal.getName());
+
+		return "redirect:/dashboard/userDetail";
+	}
+
+	@RequestMapping(value = "/userDetail/certificate", method = RequestMethod.POST)
+	public void getCertificate(final HttpServletResponse response, final Account account) {
+
+		Account bdAccount = accountDao.findUsersByUsername(account.getUsername()).get(0);
```

Comment:
#### Automated PR Comment From Polaris SAST
<!-- f89c37163bbe10e3e76449218bcdabcf -->
<!-- Comment managed by Git Commenter, do not modify! -->
### Polaris SAST Issue - SQL Injection
_High CWE-89_
Untrusted user-supplied data is inserted into a SQL statement without adequate validation, escaping, or filtering.

A user can change the intent of the SQL query, which may inappropriately disclose or corrupt data within the database.

### How to fix
Rewrite all SQL queries constructed through dynamic concatenation to use an injection-safe query mechanism such as prepared statements with parameterized queries.

Most modern programming languages provide a feature called "parameterized queries" that allow user-supplied data to be inserted safely as values in dynamic SQL queries. Rather than construct the dynamic SQL query by concatenating user-supplied data to static SQL query string fragments, data values are identified in the query by parameter markers or variables. Dynamic data is then passed through a mechanism provided by SQL that prevents the supplied data from changing the meaning of the query.

Note: the exact syntax and use of prepared statements with parameterized queries vary from language to language.